sender-stackdriver/src/main/java/zipkin2/reporter/stackdriver/StackdriverSender.java

```diff
@@ -161,8 +178,26 @@ public int messageSizeInBytes(int traceIdPrefixedSpanSize) {
     return new BatchWriteSpansCall(request.build()).map(EmptyToVoid.INSTANCE);
   }
 
+  /**
+   * Sends a malformed call to Stackdriver Trace to validate service health.
+   * @return successful status if Stackdriver Trace API responds with expected validation
+   * error (or happens to respond as success -- unexpected but okay); otherwise returns error status
+   * wrapping the underlying exception.
+   */
   @Override
   public CheckResult check() {
+
+    try {
+      healthcheckCall.clone().execute();
+    } catch (Exception e) {
+      if (e.getMessage().contains(expectedHealthcheckError)) {
+        return CheckResult.OK;
+      }
+      return CheckResult.failed(e);
+    }
+
+    // Stackdriver should not execute the malformed call successfully, so the code is unlikely to
+    // get here. But it happens, that means the server is responding, and so is minimally healthy.
```

Comment:
I think we can be more explicit for future maintainers, something like `Currently the rpc throws an exception if there is a problem, which we handle above. If we get here despite a problem, the implementation changed and we need to update this check. It's unlikely enough that we can wait and see.`.

Reply:
I'll update the wording, but should there be some logging/metrics for this? Otherwise nobody will ever notice the behavior change.